tests/h/views/panels_test.py

```diff
@@ -52,13 +53,45 @@ def test_navbar_includes_search_query(pyramid_request):
     assert result['q'] == 'tag:question'
 
 
+@pytest.mark.usefixtures('routes')
+def test_navbar_includes_search_url_when_on_user_search(pyramid_request):
+    pyramid_request.authenticated_user = None
+
+    type(pyramid_request.matched_route).name = PropertyMock(return_value='activity.user_search')
```

Comment:
I _think_ you should be able to set `request.path` in a test rather than manually mocking `matched_route`. If you've registered the route as you have below, that should work fine.